File: advanced/advanced_plugin.test.js
// Test ultra avancé clé en main pour AdvancedPlugin JS
const { AdvancedPlugin } = require('./advanced_plugin');

describe('AdvancedPlugin (JS)', () => {
  let plugin;
  beforeEach(() => {
    plugin = AdvancedPlugin;
    plugin.enabled = false;
    plugin.auditTrail = [];
  });
  test('activation admin OK', () => {
    expect(() => plugin.activate({ user: { role: 'admin' } })).not.toThrow();
    expect(plugin.enabled).toBe(true);
  });
  test('activation non-admin échoue', () => {
    expect(() => plugin.activate({ user: { role: 'guest' } })).toThrow();
    expect(plugin.enabled).toBe(true); // reste activé si déjà activé
  });
  test('run fonctionne si activé', () => {
    plugin.activate({ user: { role: 'admin' } });
    expect(plugin.run({ foo: 'bar' })).toContain('Traitement avancé');
  });
  test('run échoue si désactivé', () => {
    plugin.deactivate();
    expect(() => plugin.run({ foo: 'bar' })).toThrow();
  });
  test('auditTrail contient les actions', () => {
    plugin.activate({ user: { role: 'admin' } });
    plugin.run({ foo: 'bar' });
    expect(plugin.getAuditTrail().length).toBeGreaterThan(0);
  });
  // ... edge cases, multi-formats, etc.
});

File: advanced/advanced_plugin.test.py
# Test ultra avancé clé en main pour AdvancedPlugin Python
import unittest
from backend.components.metiers.threed.plugins.core.advanced_plugin import AdvancedPlugin

class TestAdvancedPlugin(unittest.TestCase):
    def setUp(self):
        self.plugin = AdvancedPlugin
        self.plugin.enabled = False
        self.plugin.audit_trail = []
    def test_activation_admin(self):
        self.plugin.activate({'user': {'role': 'admin'}})
        self.assertTrue(self.plugin.enabled)
    def test_activation_non_admin(self):
        with self.assertRaises(Exception):
            self.plugin.activate({'user': {'role': 'guest'}})
    def test_run_active(self):
        self.plugin.activate({'user': {'role': 'admin'}})
        result = self.plugin.run({'foo': 'bar'})
        self.assertIn('Traitement avancé', result)
    def test_run_inactive(self):
        self.plugin.deactivate()
        with self.assertRaises(Exception):
            self.plugin.run({'foo': 'bar'})
    def test_audit_trail(self):
        self.plugin.activate({'user': {'role': 'admin'}})
        self.plugin.run({'foo': 'bar'})
        self.assertGreater(len(self.plugin.get_audit_trail()), 0)
    # ... edge cases, multi-formats, etc.

if __name__ == '__main__':
    unittest.main()
